### DISCUSSION TITLE
图表渲染偶现报错「TypeError: Cannot read properties of undefined (reading 'drawShape')」

### DISCUSSION THREAD

**Discussion by licong-site** [antvis/G2#4988] [Q&A]
页面有多个图表，页面渲染偶现报错「TypeError: Cannot read properties of undefined (reading 'drawShape')」， 报错代码位置 https://github.com/antvis/G2/blob/4.2.3/src/geometry/element/index.ts#L239 。一般多刷新几次就不报错了。
报错统计这个error 今年已经出现 300+ 次了。
能否在  this.shapeFactory.drawShape()  调用前对 this.shapeFactory 的值进行判断。

![image](https://user-images.githubusercontent.com/14369099/202669997-77f28a07-8165-44b8-95d9-e6465f3463e5.png)

**Comment by hustcc** [comment]
可以来一个复现的案例代码吗？我来处理一下~
